src/sdk/namespace_s3.js

```diff
@@ -121,9 +121,9 @@ class NamespaceS3 {
     async read_object_md(params, object_sdk) {
         try {
             dbg.log0('NamespaceS3.read_object_md:', this.bucket, inspect(params));
-            const request = { Key: params.key };
+            const request = { Key: params.key, Range: 'bytes=0-4096' };
```

Comment:
Should be `0-4095` because range start and end are both inclusive.
Yes weird but that is the case - See https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35
In any case I would take it from config and then add a `-1` to the max size.